src/uwtools/drivers/forecast.py

```diff
@@ -85,45 +84,36 @@ def create_directory_structure(self, run_directory, exist_act="delete"):
            Returns: None
         '''
 
-        # Caller can only provide correct argument
+        # Caller should only provide correct argument
         if exist_act not in ["delete", "rename", "quit"]:
             raise ValueError("Bad argument to create_directory_structure")
 
         # Exit program with error if caller chooses to quit
         if exist_act == "quit":
-            logging.critical("User chose quit option")
+            logging.critical("User chose quit option when creating directory")
             sys.exit(1)
 
-        # Try to delete existing run directory if option is delete
-        try:
-            if exist_act == "delete" and os.path.isdir(run_directory):
-                shutil.rmtree(run_directory)
-        except (RuntimeError, FileExistsError) as del_error:
-            raise RuntimeError("Could not delete " +
-                               "old run directory") from del_error
-
-        # Try to rename existing run directory if caller chooses rename
-        try:
-            if exist_act == "rename" and os.path.isdir(run_directory):
-                now = datetime.now()
-                save_dir = run_directory + now.strftime("_%Y%m%d_%H%M%S")
-                shutil.move(run_directory, save_dir)
-        except (RuntimeError, FileExistsError) as rename_error:
-            raise RuntimeError("Could not rename " +
-                               "old run directory") from rename_error
+        # Delete or rename directory if it exists
+        file_helpers.handle_existing(run_directory, exist_act)
 
         # Create new run directory with two required subdirectories
         try:
+            # Create and verify new directory with subdirectories
             os.makedirs(os.path.join(run_directory, "INPUT"))
+            if not os.path.isdir(os.path.join(run_directory, "INPUT")):
+                msg = f"Directory {run_directory} with INPUT not created"
+                logging.critical(msg)
+                sys.exit(1)
             os.makedirs(os.path.join(run_directory, "RESTART"))
-            # Verify creation of new directory with subdirectories
             if not os.path.isdir(os.path.join(run_directory, "RESTART")):
-                logging.critical("New run directories not created")
+                msg = f"Directory {run_directory} with RESTART not created"
+                logging.critical(msg)
```

Comment:
This section could use a loop for reduced duplication:

```
for subdir in ("INPUT", "RESTART"):
    os.makedirs(os.path.join(run_directory, subdir))
    if not os.path.isdir(os.path.join(run_directory, subdir)):
        msg = f"Directory {run_directory} with {subdir} not created"
        logging.critical(msg)
        raise RuntimeError(msg)
```

Since it's in the `try` block, raising the error here will get handled in the except.

Reply:
Changed to a loop

Reply:
Also, file_helpers.py is in uwtools/utils